app/lib/actions/katello/repository/index_content.rb

```diff
@@ -2,21 +2,20 @@ module Actions
   module Katello
     module Repository
       class IndexContent < Actions::EntryAction
-        middleware.use Actions::Middleware::ExecuteIfContentsChanged
-
         input_format do
           param :id, Integer
-          param :dependency, Hash
-          param :contents_changed
-          param :matching_content
           param :source_repository_id
           param :full_index
         end
 
         def run
           source_repository = ::Katello::Repository.find(input[:source_repository_id]) if input[:source_repository_id]
           repo = ::Katello::Repository.find(input[:id])
-          repo.index_content(source_repository: source_repository, full_index: input[:full_index].present?)
+
+          if repo.last_contents_changed >= repo.last_indexed
```

Comment:
just a note if i review this (or someone else), we need to make sure that CV publish/promote also shows the correct content with this change